Enterprise 3.x documentation changes (initial version) (#2872)
travis-ci/docs-travis-ci-com: user/enterprise/tcie-3.x-setting-up-travis-ci-enterprise.md

---
title: Setting up Travis CI Enterprise 3.x
layout: en_enterprise
redirect_from:
  - /user/enterprise/installation/
  - /user/enterprise/prerequisites/
---

> Travis CI Enterprise works with your
> * GitHub.com or GitHub Enterprise setup
> * Bitbucket.org
> * GitLab.com

Travis CI Enteprise 3.x+ is ditributed in form of Kubernetes Off The Shell installation package, containing Helm chart and docker images for component services. It is meant to be installed as a Kubernetes managed cluster.

## Prerequisites

Travis CI Enteprrise (TCIE) installation can be executed via [Replicated KOTS](https://kots.io/) installer (for existing environment in GCE, AWS and for OpenStack managed infrastructre) or as single virtual machine installation via [Replicated kURL](https://kurl.sh/) (installs as microk8s cluster on a single instance)

  * A valid Travis CI Enterprise license or [trial license](https://enterprise.travis-ci.com/signup). The [High Availability](/user/enterprise/high-availability/) option does not require any additional trial or production license configuration.

>   * At least two virtual machines on your private cloud that meet the [system requirements](#system-requirements) - one of these must be set up as Kubernetes/ microk8s host - or Kubernetes Cluster

Author: correct, rephrasing to `Two virtual machines on your private cloud that meet the [system requirements](#system-requirements) - one of these must be set up as Kubernetes/ microk8s host, second as Worker Machine host - or Kubernetes cluster` in subsequent commit

    - *Homepage URL* - `https://<your-travis-ci-enterprise-domain>`
    - *Authorization callback URL* - `https://<your-travis-ci-enterprise-domain>/api`

    URLs must include https or http at the beginning and cannot have trailing slashes.
    

### System Requirements

The standard setup consists of the *Travis CI Enterprise
Platform* which hosts Kubernetes cluster with each service in separate pod, and one or more
*Worker hosts* which run the tests/jobs in isolated containers using LXC
and Docker.

Each dedicated host or hypervisor (VMWare, OpenStack, or EC2) should run **Ubuntu 18.04** and have at least **8 gigs of RAM and 4 CPUs**. If the same host is to run as Worker host, inimum requirement would be **16 gigs of RAM and 8 CPUs**.

If you're running on EC2, we recommend the **c4.2xlarge** instance type for both **Platform** and **Worker**. We also recommend using an image that uses EBS for the root volume, as well as allocating 80 gigs of space to it.

> For [high availability (HA)](/user/enterprise/high-availability/) configurations, you will also need self-hosted:

Author: Correct, rephrased to `you will also need to configure appropriately` in subsequent commit

* [Redis](https://redis.io/),
* [RabbitMQ](https://www.rabbitmq.com/)
* [Postgres](https://www.postgresql.org/)

## 1. Setting up Enterprise Platform 

The Travis CI Enterprise Platform handles licensing, coordinates worker
processes, and maintains the Enterprise user and admin dashboard. It must be
installed as Kubernetes cluster or on at least one machine instance acting as microk8s host 
or multiple instances managed by Kubernetes, separate from that of the Travis CI
Enterprise worker. We recommend **compute optimized** instance running
Ubuntu 18.04 LTS or later as the underlying operating system.

> If you are migrating from Travis CI enteprise 2.x, please make sure to read and execute the [datbase migration instructions](/user/enterprise/tcie-3.x-migrating-db-from-2.x-to-3.x./) first.

1. *On your infrastructure management platform*, create a Travis CI Platform Security Group.

>     If you're setting up your instance image or Kubernetes cluster for the first time, you most probably need to create some kind of

Reviewer: ```suggestion
    If you're setting up your instance image or Kubernetes cluster for the first time, you need to create 
```

    a Security Group. Create an entry for each port in the table below:

    | Port | Service         | Description                                                                  |
    |:-----|:----------------|:-----------------------------------------------------------------------------|
    | 8800 | Custom TCP Rule | This port is to access the admin dashboard for your Enterprise installation. |
    | 5672 | Custom TCP Rule | For RabbitMQ Non-SSL.                                                        |
    | 4567 | Custom TCP Rule | For RabbitMQ SSL.                                                            |
    | 3333 | HTTPS           | TCIE 3.x User administration Web application over HTTPS access.              |
    | 443  | HTTPS           | Web application over HTTPS access.                                           |
    | 80   | HTTP            | Web application access.                                                      |

>     | 22   | SSH             | SSH access.                                                                  |

Reviewer: Please add another column explaining from where this traffic is coming from to ensure that users know how to securely configure their Firewalls / Security Groups

Author: Added more details to the description (inbound access) in subsequent commit

> 2. If you're using a hostname and not just an IP address, this would be a time to configure it.

Reviewer: ```suggestion
2. Please configure your hostname now. (Skip if you will access TCIE through an IP address)
```

> 3. Set-up your cluster/machine instances configuration - at least 1 virtual machine is needed (in microk8s scenario) or Kubernetes cluster prepared and started

Reviewer: ```suggestion
3. Set up your cluster/machine instances configuration - at least 1 virtual machine is needed (in microk8s scenario) or Kubernetes cluster prepared and started
```

4. On *your local machine*, assuming the proper connection with the cluster exists, install Replicated KOTS.
5. TCIE 3.x installed on **single** virtual machine instance
    1. On *your new virtual machine instance* run kurl.sh via 'curl https://kurl.sh/latest | sudo bash' 

>     2. Get credentials from microk8s cluster running on *your new vm instance*

Reviewer: How does that work? Do I need to run a command for that?

Author: Added redirection to kurls.sh docs in susbequent commit

    3. Please refer to various cluster and installation options in [kURL documentation](https://kurl.sh/docs/install-with-kurl/). Please adjust your setup and configuration according to your needs before progressing with TCIE 3.X installation
6. TCIE 3.x installed as Kubernetes cluster in the cloud

>     1. Connect to your cluster and get it's configuration, particularly credentials

Reviewer: ```suggestion
    1. Connect to your cluster and get its configuration, particularly credentials
```

How does this work? What command do I need to run to get credentials?

Author: Adressed by rephrasing the sentence to be more exact in subsequent commit

5. Run `kubectl kots install tci-enterprise-kots` to install TCIE 3.x. Please note down administrative password and namespace used during this step.
6. *In your browser*, navigate to `http://localhost:8800` to complete the setup. The TCIE 3.x admin console will be automatically enabled during first installation :

   1. Authorize with a password set during the installation process 
   2. Upload your Travis CI Enterprise license. 

>    3. Connect your GitHub Enterprise or GitHub.com (or other VCS).

Reviewer: ```suggestion
   3. Connect your Source Control System ( GitHub Enterprise Server, GitHub.com, or Bitbucket).
```

   4. Optionally, configure Email, Metrics and Caches.
   5. Copy the *RabbitMQ password* for the Worker setup.

> If you have decided to use a self-signed certificate or wish to use Let's Encrypt certificate, there may be additional configuration steps required. Please see our page on [SSL Certificate Management](/user/enterprise/ssl-certificate-management) for more information.

### 1.1 Example TCIE 3.x installation for GCE on MacOSX

#### 1.1.1 Python

You'll want to make sure you have the latest version of Python 3, if you're on Mac OSX (a clean install) you'll want to install Homebrew via:

```bash
ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)
```

Once brew is up and running, you'll install Python via: 

```bash
brew install python
```

Now check if Python is installed via running: 

```bash
python -version
``` 

#### 1.1.2 Google SDK

Install the Google Cloud SDK via: 

```bash
curl https://sdk.cloud.google.com | bash
```

> If you try to run at this point ```gcloud init``` it will not work, you must restart your shell via:

Reviewer: ```suggestion
Please restart your shell session before running `gcloud init`: 
```
